### ISSUE TITLE
[python/php/java] Move SslSocket impl back to sys.ssl package

### ISSUE THREAD

**Issue by kevinresol** [HaxeFoundation/haxe#7542]
They currently sits in:

- `python.net.SslSocket`
- `php.net.SslSocket`
- `java.net.SslSocket`

**Comment by Simn** [comment]
I don't think this is 4.0 priority, but I'm also not quite sure what this is about.

**Comment by kevinresol** [comment]
If I understand correctly, these implementations are actually compatible with sys.ssl.Socket, moving them back to that package saves user from the conditional compilation mess.

Maybe just a typedef is fine.

**Comment by Simn** [comment]
I agree, but that doesn't sound like it's high priority. Moving to 4.1 for now.

**Comment by barisyild** [comment]
the same problem exists in the csharp target.

**Comment by kevinresol** [comment]
The problem is: C# doesn't have a SslSocket implementation: https://github.com/HaxeFoundation/haxe/issues/7543
